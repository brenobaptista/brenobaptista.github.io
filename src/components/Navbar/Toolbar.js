import React from 'react';
import PropTypes from 'prop-types';

import Logo from './Logo';
import NavigationItems from './NavigationItems';
import DrawerToggle from './DrawerToggle';

const toolbar = (props) => {
  const { drawerToggleClicked } = props;

  return (
    <header className="toolbar">
      <DrawerToggle clicked={drawerToggleClicked} />
      <div className="logoToolbar">
        <Logo />
      </div>
      <nav className="desktopOnly">
        <NavigationItems />
      </nav>
    </header>
  );
};

toolbar.propTypes = {
  drawerToggleClicked: PropTypes.func,
};

toolbar.defaultProps = {
  drawerToggleClicked: () => {},
};

export default toolbar;
