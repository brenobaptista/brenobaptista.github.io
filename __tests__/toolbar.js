import React from 'react'
import { render } from '@testing-library/react'
import Toolbar from '../src/components/Navbar/Toolbar'

test('matches snapshot', () => {
  const clicked = jest.fn()
  const { asFragment } = render(<Toolbar drawerToggleClicked={clicked} email='' />)

  expect(asFragment(<Toolbar drawerToggleClicked={clicked} email='' />)).toMatchSnapshot()
})
