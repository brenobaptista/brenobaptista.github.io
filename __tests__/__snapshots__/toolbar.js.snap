// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="toolbar"
  >
    <div
      aria-label="toggle sidedrawer"
      class="drawerToggle"
      role="button"
      tabindex="0"
    >
      <div />
      <div />
      <div />
    </div>
    <div
      class="logoToolbar"
    >
      <div
        class="logo"
      >
        <img
          alt="logo"
          src="test-file-stub"
        />
      </div>
    </div>
    <nav
      class="desktopOnly"
    >
      <ul
        class="navigationItems"
      >
        <li
          class="navigationItem"
        >
          <a
            href="/"
          >
            HOME
          </a>
        </li>
        <li
          class="navigationItem"
        >
          <a
            href="/projects"
          >
            PROJECTS
          </a>
        </li>
        <li
          class="navigationItem"
        >
          <a
            href="/blog"
          >
            BLOG
          </a>
        </li>
      </ul>
    </nav>
  </header>
</DocumentFragment>
`;
